core/src/main/java/org/sonarsource/sonarlint/core/smartnotifications/SmartNotifications.java

```diff
@@ -162,16 +163,15 @@ private static List<ServerNotification> retrieveServerNotifications(DevelopersAp
   public void onServerEventReceived(SonarServerEventReceivedEvent eventReceived) {
     var serverEvent = eventReceived.getEvent();
     if (serverEvent instanceof SmartNotificationEvent) {
-      notifyClient((SmartNotificationEvent) serverEvent);
+      notifyClient(eventReceived.getConnectionId(), (SmartNotificationEvent) serverEvent);
     }
   }
 
-  private void notifyClient(SmartNotificationEvent event) {
+  private void notifyClient(String connectionId, SmartNotificationEvent event) {
     var projectKey = event.getProject();
-    configurationRepository.getBoundScopesByProject(projectKey).stream()
-      .collect(Collectors.groupingBy(BoundScope::getConnectionId))
-      .forEach((connectionId, scope) -> client.showSmartNotification(new ShowSmartNotificationParams(event.getMessage(), event.getLink(),
-        scope.stream().map(BoundScope::getId).collect(Collectors.toSet()), event.getCategory(), connectionId)));
+    var boundScopes = configurationRepository.getBoundScopesToConnectionAndSonarProject(connectionId, projectKey);
+    client.showSmartNotification(new ShowSmartNotificationParams(event.getMessage(), event.getLink(),
+      boundScopes.stream().map(BoundScope::getConfigScopeId).collect(Collectors.toSet()), event.getCategory(), connectionId));
```

Comment:
Same here, we notify for every child scope, so it will be up to the client to filter and keep only the parent.